### PR TITLE
COM-345: Add `helperIcon` prop to MenuItemGroup

### PR DESCRIPTION
**Pull request by jomunker**
Depends on #1233.
 
This PR adds the option to add a helper icon (+ helper Text) to menu group sections.

- [x] Add changeset

### helperIcon + undefined helperText
Screenshot 2024-01-26 at 17 14 31
### helperText + undefined helperIcon
Screenshot 2024-01-26 at 17 13 55
### helperText and helperIcon defined
Screenshot 2024-01-26 at 17 14 19

### CODE REVIEW

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/ItemGroup.tsx:56]: Instead of having two props, why not add one `helper?: ReactNode` and let developers decide what they want to render?

**Review by johnnyomair — COMMENTED** [review]
We'll probably need this in 1 out of 100 instances... Couldn't this be solved via `title`? Of course, getting the initials wouldn't work then, but thats what `shortTitle` is for 🙂

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx:56]: Because we can't expect a dev to know this workaround 🤷🏻‍♂️ 
image

Comment (johnnyomair) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx:56]: image

I just tried this in the MUI docs, and it works perfectly fine with the icon?

**Comment by johnnyomair** [comment]
> We could think about allowing the icon to be a `React.ReactNode` so that it can be used otherwis

I'd prefer that. Would also be more in line with MUI's API design. For instance, Button has a prop `endIcon` that's a ReactNode, Input has `endAdornment` that's a ReactNode, etc. Our APIs should be similarly flexible.

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx:56]: With MUI icons but not with our custom ones...

Comment (johnnyomair) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx]: Then we should fix that 😉 

https://github.com/vivid-planet/comet/pull/1663

File: .changeset/wise-moose-argue.md
Comment (johnnyomair) [inline .changeset/wise-moose-argue.md:5]:
```suggestion
Add `helperIcon` prop to MenuItemGroup. Its intended purpose is to render an icon with a `Tooltip` behind the group section title, if the menu is not collapsed.
```

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/ItemGroup.tsx:113]: This isn't part of this PR, right?

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/ItemGroup.tsx:47]: This isn't used anymore

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/ItemGroup.tsx:30]: Shouldn't this be part of the helper icon?

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/ItemGroup.tsx:10]: This class key isn't needed anymore

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx:113]: how is this a change? 🤔 Maybe because of a rebase.

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx:30]: imo passing an icon without styles should result in an acceptable outcome. And since the margin doesn't hurt on the title I'd keep it

Comment (thomasdax98) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx:30]: You could add the margin to the icon if one exists. Should work like this:

```
titleMenuOpen: {
    [`& + .${svgIconClasses.root}`]: {
        marginLeft: theme.spacing(1),
    },
}
```

But that's a minor detail IMO

**Review by thomasdax98 — APPROVED** [review]

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx
Comment (johnnyomair) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx:30]: We can keep it like it currently is.

**Review by johnnyomair — DISMISSED** [review]

**Comment by johnnyomair** [comment]
We should only merge this once #1233 is merged.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**
